ecosystem/indexer-grpc/indexer-test-transactions/build.rs

```diff
@@ -1,74 +1,123 @@
 // Copyright (c) Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-// build.rs
 use std::{env, fs, path::Path};
 
-fn main() {
-    let out_dir = env::var("OUT_DIR").unwrap();
-    let dest_path = Path::new(&out_dir).join("generate_transactions.rs");
-
-    let mut all_transactions_code = String::new();
-
-    create_directory_if_missing("json_transactions/imported_mainnet_txns");
-    create_directory_if_missing("json_transactions/imported_testnet_txns");
-    create_directory_if_missing("json_transactions/scripted_transactions");
-
-    all_transactions_code.push_str(&process_directory(
-        "imported_mainnet_txns",
-        "imported_mainnet_txns",
-    ));
-    all_transactions_code.push_str(&process_directory(
-        "imported_testnet_txns",
-        "imported_testnet_txns",
-    ));
-    all_transactions_code.push_str(&process_directory(
-        "scripted_transactions",
-        "scripted_transactions",
-    ));
-
-    fs::write(dest_path, all_transactions_code).unwrap();
+const IMPORTED_MAINNET_TXNS: &str = "imported_mainnet_txns";
+const IMPORTED_TESTNET_TXNS: &str = "imported_testnet_txns";
+const SCRIPTED_TRANSACTIONS_TXNS: &str = "scripted_transactions";
+#[derive(Default)]
+pub struct TransactionCodeBuilder {
+    transactions_code: String,
+    name_function_code: String,
```

Comment:
What are these strings used for?

Reply:
The `transactions_code` and `name_function_code` are used to create code snippets that are generated dynamically based on the transaction jsons directories, which will be used as an input of processor tests and a util function to map the name of variable to file_name. 

`transactions_code` is used to collect all the generated Rust code for the transaction constants. It stores the code for pub const declarations that point to JSON files in the given directories.

`name_function_code` is used to build the matching logic for a function that takes a transaction's constant data and returns its corresponding name. This function will be generated only for the scripted txn.

Reply:
Got it, so will this basically replace importing the test-transactions crate in the processors repo? How is this going to be used and how does this improve devex?

Reply:
test-trasnaction crate is to support local testing. This won't be run explicitly , but this will be used when we update the dependency in processor-repo with the latest commit hash in Cargo.toml. 

```
aptos-indexer-test-transactions = { git = "https://github.com/aptos-labs/aptos-core.git", rev = "7246f0536d599789d7dd5e9fb776554abbe11eac" }
```